File: frontend/src/types/seal.ts
/**
 * 电子签章管理类型定义
 * 对应后端: /v1/seals
 */

/** 印章类型 */
export type SealType = 'official' | 'finance' | 'contract' | 'legal' | 'custom'

/** 印章状态 */
export type SealStatus = 'active' | 'inactive' | 'revoked'

/** 印章使用场景 */
export type SealScene = 'hr_contract' | 'purchase_contract' | 'electronic_contract'

/** 印章信息 */
export interface SealInfo {
  /** 印章ID */
  sealId: string
  /** 印章名称 */
  sealName: string
  /** 印章类型 */
  sealType: SealType
  /** 印章图片（Base64） */
  sealImage: string
  /** 状态 */
  status: SealStatus
  /** 保管人 */
  keeper: string
  /** 授权使用人ID列表 */
  authorizedUsers: string[]
  /** 授权使用场景 */
  authorizedScenes: SealScene[]
  /** 创建时间 */
  createTime: string
  /** 更新时间 */
  updateTime: string
  /** 创建人 */
  createBy: string
  /** 备注 */
  remark?: string
}

/** 印章使用记录 */
export interface SealUsageLog {
  /** 记录ID */
  logId: string
  /** 印章ID */
  sealId: string
  /** 印章名称（冗余，便于展示） */
  sealName: string
  /** 业务类型 */
  businessType: SealScene
  /** 业务ID（合同ID） */
  businessId: string
  /** 业务编号（合同编号，便于展示） */
  businessNo: string
  /** 操作人 */
  operator: string
  /** 操作时间 */
  operateTime: string
  /** IP地址 */
  ipAddress: string
  /** 备注 */
  remark?: string
}

/** 印章表单数据（新增/编辑） */
export interface SealFormData {
  sealName: string
  sealType: SealType
  sealImage: string
  keeper: string
  authorizedUsers: string[]
  authorizedScenes: SealScene[]
  remark?: string
}

/** 印章查询表单 */
export interface SealQueryForm {
  keyword?: string
  sealType?: SealType | null
  status?: SealStatus | null
  page?: number
  size?: number
}

/** 印章使用记录查询表单 */
export interface SealUsageLogQueryForm {
  sealId?: string
  businessType?: SealScene | null
  page?: number
  size?: number
}

/** 印章类型选项 */
export const SealTypeOptions = [
  { label: '公章', value: 'official' as const },
  { label: '财务专用章', value: 'finance' as const },
  { label: '合同专用章', value: 'contract' as const },
  { label: '法人章', value: 'legal' as const },
  { label: '自定义', value: 'custom' as const },
]

/** 印章状态选项 */
export const SealStatusOptions = [
  { label: '启用', value: 'active' as const },
  { label: '停用', value: 'inactive' as const },
  { label: '作废', value: 'revoked' as const },
]

/** 印章使用场景选项 */
export const SealSceneOptions = [
  { label: '人事合同', value: 'hr_contract' as const },
  { label: '采购合同', value: 'purchase_contract' as const },
  { label: '电子合同', value: 'electronic_contract' as const },
]

/** 印章类型标签映射 */
export const SealTypeLabelMap: Record<SealType, string> = {
  official: '公章',
  finance: '财务专用章',
  contract: '合同专用章',
  legal: '法人章',
  custom: '自定义',
}

/** 印章状态标签映射 */
export const SealStatusLabelMap: Record<SealStatus, string> = {
  active: '启用',
  inactive: '停用',
  revoked: '作废',
}

/** 印章使用场景标签映射 */
export const SealSceneLabelMap: Record<SealScene, string> = {
  hr_contract: '人事合同',
  purchase_contract: '采购合同',
  electronic_contract: '电子合同',
}

File: backend/src/main/java/com/foodtraceability/entity/seal/Seal.java
package com.foodtraceability.entity.seal;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 印章实体类
 * 存储印章基本信息、授权使用人和授权使用场景
 *
 * <p>枚举值（与前端 types/seal.ts 保持一致）：
 * <ul>
 *   <li>sealType: official-公章 / finance-财务专用章 / contract-合同专用章 / legal-法人章 / custom-自定义</li>
 *   <li>status: active-启用 / inactive-停用 / revoked-作废</li>
 *   <li>authorizedScenes: hr_contract-人事合同 / purchase_contract-采购合同 / electronic_contract-电子合同</li>
 * </ul>
 *
 * <p>JSON 字段说明：
 * authorizedUsers / authorizedScenes 使用 String 存储 JSON 数组字符串，
 * 序列化/反序列化由 Service 层负责（如 ["U001","U002"]）。
 */
@TableName("seals")
public class Seal implements Serializable {

    private static final long serialVersionUID = 1L;

    /** 印章ID */
    @TableId(value = "seal_id", type = IdType.ASSIGN_ID)
    private String sealId;

    /** 印章名称 */
    @TableField("seal_name")
    private String sealName;

    /** 印章类型(official/finance/contract/legal/custom) */
    @TableField("seal_type")
    private String sealType;

    /** 印章图片(Base64/data URI) */
    @TableField("seal_image")
    private String sealImage;

    /** 状态(active/inactive/revoked) */
    @TableField("status")
    private String status;

    /** 保管人 */
    @TableField("keeper")
    private String keeper;

    /** 授权使用人ID列表(JSON数组字符串,如 ["U001","U002"]) */
    @TableField("authorized_users")
    private String authorizedUsers;

    /** 授权使用场景列表(JSON数组字符串,如 ["hr_contract","purchase_contract"]) */
    @TableField("authorized_scenes")
    private String authorizedScenes;

    /** 创建人 */
    @TableField("create_by")
    private String createBy;

    /** 备注 */
    @TableField("remark")
    private String remark;

    /** 逻辑删除标记(0-未删除 1-已删除) */
    @TableLogic
    @TableField("deleted")
    private Integer deleted;

    /** 创建时间 */
    @TableField(value = "create_time", fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /** 更新时间 */
    @TableField(value = "update_time", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    // ==================== 状态常量 ====================

    /** 启用 */
    public static final String STATUS_ACTIVE = "active";
    /** 停用 */
    public static final String STATUS_INACTIVE = "inactive";
    /** 作废 */
    public static final String STATUS_REVOKED = "revoked";

    // ==================== 印章类型常量 ====================

    /** 公章 */
    public static final String TYPE_OFFICIAL = "official";
    /** 财务专用章 */
    public static final String TYPE_FINANCE = "finance";
    /** 合同专用章 */
    public static final String TYPE_CONTRACT = "contract";
    /** 法人章 */
    public static final String TYPE_LEGAL = "legal";
    /** 自定义 */
    public static final String TYPE_CUSTOM = "custom";

    // ==================== 使用场景常量 ====================

    /** 人事合同 */
    public static final String SCENE_HR_CONTRACT = "hr_contract";
    /** 采购合同 */
    public static final String SCENE_PURCHASE_CONTRACT = "purchase_contract";
    /** 电子合同 */
    public static final String SCENE_ELECTRONIC_CONTRACT = "electronic_contract";

    // ==================== Getter & Setter ====================

    public String getSealId() {
        return sealId;
    }

    public void setSealId(String sealId) {
        this.sealId = sealId;
    }

    public String getSealName() {
        return sealName;
    }

    public void setSealName(String sealName) {
        this.sealName = sealName;
    }

    public String getSealType() {
        return sealType;
    }

    public void setSealType(String sealType) {
        this.sealType = sealType;
    }

    public String getSealImage() {
        return sealImage;
    }

    public void setSealImage(String sealImage) {
        this.sealImage = sealImage;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getKeeper() {
        return keeper;
    }

    public void setKeeper(String keeper) {
        this.keeper = keeper;
    }

    public String getAuthorizedUsers() {
        return authorizedUsers;
    }

    public void setAuthorizedUsers(String authorizedUsers) {
        this.authorizedUsers = authorizedUsers;
    }

    public String getAuthorizedScenes() {
        return authorizedScenes;
    }

    public void setAuthorizedScenes(String authorizedScenes) {
        this.authorizedScenes = authorizedScenes;
    }

    public String getCreateBy() {
        return createBy;
    }

    public void setCreateBy(String createBy) {
        this.createBy = createBy;
    }

    public String getRemark() {
        return remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }

    public Integer getDeleted() {
        return deleted;
    }

    public void setDeleted(Integer deleted) {
        this.deleted = deleted;
    }

    public LocalDateTime getCreateTime() {
        return createTime;
    }

    public void setCreateTime(LocalDateTime createTime) {
        this.createTime = createTime;
    }

    public LocalDateTime getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(LocalDateTime updateTime) {
        this.updateTime = updateTime;
    }
}
